### PR TITLE
DATAUP-314: Jobs and job data-related changes:

### PR DESCRIPTION
**Pull request by ialarmedalien**
# Description of PR purpose/changes

This PR extracts some of the jobs-related changes from existing PR #2105 and combines them with some other job data-related changes; adding all these changes to 2105 would have made the PR too large, so once this change set is merged, I will rebase 2105, which should simplify it quite a bit.

Changes from 2105:
- alter naming in Jobs module to be internally and externally consistent wrt: job status and job state objects
- remove unneeded job-state.scss file, move contents to job-status.scss
- update some spec files to use newer JS conventions

New changes:
- use the JobsData module to populate the jobs in the `TestAppObj` used for testing and `testAppObj-ci.js` used for the bulk import cell; unfortunately, due to the way that assets are served (can't/shouldn't serve data from `/test/` to the browser), this means that there are two copies of `jobsData.js`, but given that we will be able to remove `testAppObj-ci.js` once the bulk import cell is complete, this isn't too big a deal.

The alternative would be for the tests to use `jobsData.js` from `nbextensions/bulkImportCell`, and then move that file to the test data dir once the bulk import cell is complete. I don't have strong feelings either way if that is your preferred implementation.

# Jira Ticket / Issue #
e.g. https://kbase-jira.atlassian.net/browse/DATAUP-314
- [x] Added the Jira Ticket to the title of the PR (e.g. `DATAUP-69 Adds a PR template`)

# Testing Instructions
* Details for how to test the PR:
- [ ] Tests pass locally and in GitHub Actions -- **Integration tests are failing, unit tests are passing**
- [x] Changes available by spinning up a local narrative and navigating to see that nothing has changed!

# Dev Checklist:

- [x] My code follows the guidelines at https://sites.google.com/lbl.gov/trussresources/home?authuser=0
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] (JavaScript) I have run Prettier and ESLint on changed code manually or with a git precommit hook

### CODE REVIEW

File: kbase-extension/scss/partials/_job-status.scss [hunk 1]
```diff
@@ -53,7 +53,28 @@
       color: use-color('info');
     }
   }
-// panel-heading border none
+
```
Comment (ialarmedalien) [inline kbase-extension/scss/partials/_job-status.scss:56]: consolidate job state/status scss files

File: kbase-extension/static/kbase/js/common/jobs.js [hunk 2]
```diff
@@ -1,37 +1,129 @@
...
+            'terminated',
+            'does_not_exist',
+        ];
+
+    /**
+     * A jobState object is deemed valid if
+     * 1. It's an object (not an array or atomic type)
+     * 2. It has a 'created' key
+     * 3. It has a 'job_id' key
+     * 4. It has a 'status' key, which appears in the array validJobStates
```
Comment (ialarmedalien) [inline kbase-extension/static/kbase/js/common/jobs.js:30]: new check

File: kbase-extension/static/kbase/js/common/jobs.js [hunk 3]
```diff
@@ -1,37 +1,129 @@
...
+        try {
+            if (
+                'job_id' in jobInfo &&
+                Object.prototype.toString.call(jobInfo.job_params[0]) === '[object Object]' &&
+                Object.keys(jobInfo.job_params[0]).length > 0
+            ) {
+                return true;
+            }
+        // eslint-disable-next-line no-empty
+        } catch (err) {}
```
Comment (ialarmedalien) [inline kbase-extension/static/kbase/js/common/jobs.js:91]: we're just checking whether the job info object is valid; we don't really care how it deviates from the expected structure, so this error can be ignored

File: kbase-extension/static/kbase/js/common/jobs.js [hunk 4]
```diff
@@ -79,11 +178,36 @@ define([
...
     }
 
+    /**
+     * Ensures that the child_jobs data is in the correct place in the jobState object
+     *
+     * It should be possible to remove this function after migrating to ee 2.5
+     * @param {object} jobState
+     * @return {object} jobState, possibly edited
+     */
+    function updateJobModel(jobState) {
```
Comment (ialarmedalien) [inline kbase-extension/static/kbase/js/common/jobs.js:196]: temporary hack -- once the backend is updated to use the new format for jobs data, this function can disappear

File: kbase-extension/static/kbase/js/common/jobs.js [hunk 5]
```diff
@@ -134,10 +257,15 @@ define([
             }
 
             jobLines.push(
-                'Finished with ' + niceState(jobState.status) + ' at '
-                + span({
-                    class: 'kb-timestamp'
-                }, format.niceTime(jobState.finished))
+                'Finished with ' +
```
Comment (ialarmedalien) [inline kbase-extension/static/kbase/js/common/jobs.js:260]: prettier

File: kbase-extension/static/kbase/js/common/jobs.js [hunk 6]
```diff
@@ -163,19 +291,29 @@ define([
...
-    function runTime (jobState) {
+    function runTime(jobState) {
         if (!jobState.finished) {
-            return UI.loading({ color: 'green' })
-                + ' Started running job at '
-                + span({
-                    class: 'kb-timestamp'
-                }, format.niceTime(jobState.running))
-                + span({ class: 'runClock', dataElement: 'clock' });
+            return (
```
Comment (ialarmedalien) [inline kbase-extension/static/kbase/js/common/jobs.js:296]: prettier

File: kbase-extension/static/kbase/js/common/jobs.js [hunk 7]
```diff
@@ -189,52 +327,42 @@ define([
...
+            UI.loading({ color: 'orange' }) +
+            ' In the queue since ' +
+            span(
+                {
+                    class: 'kb-timestamp',
+                },
+                format.niceTime(jobState.created)
+            );
 
-    /**
```
Comment (ialarmedalien) [inline kbase-extension/static/kbase/js/common/jobs.js:214]: moved to top of file

File: nbextensions/bulkImportCell/jobsData.js [hunk 8]
```diff
@@ -0,0 +1,293 @@
+define(['common/format'], (format) => {
```
Comment (ialarmedalien) [inline nbextensions/bulkImportCell/jobsData.js:1]: This populates the job status table with a set of jobs, one in each state. The same data is used for tests and when running the bulk import cell on CI.

File: nbextensions/bulkImportCell/testAppObj-ci.js [hunk 9]
```diff
@@ -0,0 +1,472 @@
+define(['./jobsData'], (JobsData) => {
```
Comment (ialarmedalien) [inline nbextensions/bulkImportCell/testAppObj-ci.js:1]: Convert the old JSON file to JS, and populate the child jobs using the data from `jobsData.js`

File: test/data/testAppObj.js [hunk 10]
```diff
@@ -0,0 +1,474 @@
+define(['./jobsData'], (JobsData) => {
```
Comment (ialarmedalien) [inline test/data/testAppObj.js:1]: identical to the data in `testAppObj-ci.js`

File: test/unit/spec/common/cellComponents/tabs/infoTab-Spec.js [hunk 11]
```diff
@@ -1,111 +1,93 @@
...
-/*global define*/ // eslint-disable-line no-redeclare
-/*global describe, expect, it*/
-/*jslint white: true*/
-
-define([
-    'common/cellComponents/tabs/infoTab'
-], function(
-    infoTabWidget
-) {
+define(['common/cellComponents/tabs/infoTab'], (infoTabWidget) => {
```
Comment (ialarmedalien) [inline test/unit/spec/common/cellComponents/tabs/infoTab-Spec.js:1]: All changes are from eslint/prettier

File: test/unit/spec/common/cellComponents/tabs/jobStateList-Spec.js [hunk 12]
```diff
@@ -70,7 +71,7 @@ define([
                 expect($(node).find(item).length).toBeGreaterThan(0);
             });
 
-            expect($(node).find('.kb-job-status__row').length).toEqual(7);
+            expect($(node).find('.kb-job-status__row').length).toEqual(JobsData.allJobs.length);
```
Comment (ialarmedalien) [inline test/unit/spec/common/cellComponents/tabs/jobStateList-Spec.js:74]: remove magic number

File: nbextensions/bulkImportCell/testAppObj-ci.js [hunk 13]
```diff
@@ -0,0 +1,487 @@
...
+    'use strict';
+
+    const jobData = JobsData.allJobs;
+    const jobRemapping = {
```
Comment (ialarmedalien) [inline nbextensions/bulkImportCell/testAppObj-ci.js:5]: These jobIDs are from the narrative https://ci.kbase.us/narrative/57559 / "Many cell types" on localhost.

File: test/unit/spec/common/jobsSpec.js [hunk 14]
```diff
@@ -1,231 +1,97 @@
...
+    const jobsModuleExports = [
+        'createJobStatusLines',
+        'jobAction',
+        'jobLabel',
+        'niceState',
+        'validJobStates',
+        'isValidJobStateObject',
+    ];
+
+    // console.log(JobsData.allJobs)
```
Comment (briehl) [inline test/unit/spec/common/jobsSpec.js:17]: Don't need the comment.

**Review by briehl — APPROVED** [review]
LGTM! :tada:

File: test/unit/spec/common/jobsSpec.js (hunk 14 above)
Comment (ialarmedalien) [reply test/unit/spec/common/jobsSpec.js:17]: It's removed in #2105

**Pull request merged**